### PR TITLE
wireguard: remove deprecated userspace fallback

### LINKED ISSUE #33594: [v1.17] wireguard: remove deprecated userspace fallback
https://github.com/cilium/cilium/pull/31867 deprecated the userspace fallback. We can remove it in `v1.17`.

### PR DESCRIPTION
**Pull request by julianwiedmann**
The userspace fallback was deprecated in v1.16 with https://github.com/cilium/cilium/pull/31867. Let's remove it now.

Fixes: https://github.com/cilium/cilium/issues/33594

### CODE REVIEW

**Review by gandro — APPROVED** [review]
Great diff! image

Thank you for taking care of this! :pray:

File: pkg/wireguard/agent/agent.go [hunk 1]
```diff
@@ -168,60 +164,6 @@ func (a *Agent) InitLocalNodeFromWireGuard(localNode *node.LocalNode) {
...
-
-	scopedLog := log.WithField(logfields.LogSubsys, "wireguard-userspace")
-	logger := &device.Logger{
-		Verbosef: scopedLog.Debugf,
-		Errorf:   scopedLog.Errorf,
-	}
-	dev := device.NewDevice(tundev, conn.NewDefaultBind(), logger)
-
-	// cleanup removes the tun device and uapi socket
-	a.cleanup = append(a.cleanup, func() {
```
Comment (gandro) [inline pkg/wireguard/agent/agent.go:200]: On closer inspection, let's also remove `a.cleanup` since it is no longer used

**Review by marseel — APPROVED** [review]

**Review by derailed — APPROVED** [review]
@julianwiedmann Nice work!

File: pkg/wireguard/agent/agent.go (hunk 1 above)
Comment (julianwiedmann) [reply pkg/wireguard/agent/agent.go:200]: Pushed https://github.com/cilium/cilium/compare/cc5725858eb8a5576a37ca386666e01f88d802d6..573368a5283dfe579798b4d91e50039acdd77deb.

**Comment by julianwiedmann** [comment]
@learnitall ping :pray:

**Review by learnitall — APPROVED** [review]
Thank you for the ping, LGTM :rocket:

**Pull request merged**